.github/workflows/publish.yaml

```
@@ -0,0 +1,152 @@
+name: Publish
+run-name: Publish Repository Actions 🛫
+on:
+  release:
+    types:
+      - published
+    branches:
+      - master
+      - 'release/**'
+permissions:
+  contents: read
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+jobs:
+  check-tag:
+    name: Check Release Tag
+    if: ${{ github.repository_owner == 'obsproject' }}
+    runs-on: ubuntu-22.04
+    outputs:
+      validTag: ${{ steps.check.outputs.validTag }}
+      flatpakMatrix: ${{ steps.check.outputs.flatpakMatrix }}
+    steps:
+      - name: Check Release Tag ☑️
+        id: check
+        run: |
+          : Check Release Tag ☑️
+          if [[ "${RUNNER_DEBUG}" ]]; then set -x; fi
+          shopt -s extglob
+
+          case "${GITHUB_REF_NAME}" in
+            +([0-9]).+([0-9]).+([0-9]) )
+              echo 'validTag=true' >> $GITHUB_OUTPUT
+              echo 'flatpakMatrix=["beta", "stable"]' >> $GITHUB_OUTPUT
+              ;;
+            +([0-9]).+([0-9]).+([0-9])-@(beta|rc)*([0-9]) )
+              echo 'validTag=true' >> $GITHUB_OUTPUT
+              echo 'flatpakMatrix=["beta"]' >> $GITHUB_OUTPUT
+              ;;
+            *) echo 'validTag=false' >> $GITHUB_OUTPUT ;;
+          esac
+
+  flatpak-publish:
+    name: Publish to Flathub 📦
+    needs: check-tag
+    if: ${{ github.repository_owner == 'obsproject' && fromJSON(needs.check-tag.outputs.validTag) }}
+    runs-on: ubuntu-22.04
+    defaults:
+      run:
+        shell: bash
+    env:
+      FLATPAK_BUILD_PATH: flatpak_app/files/share
+    container:
+      image: bilelmoussaoui/flatpak-github-actions:kde-6.4
+      options: --privileged
+    strategy:
+      matrix:
+        branch: ${{ fromJSON(needs.check-tag.outputs.flatpakMatrix) }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          fetch-depth: 0
+          set-safe-directory: ${{ github.workspace }}
+
+      - name: Set Up Environment 🔧
+        id: setup
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          : Set Up Environment 🔧
+          if [[ "${RUNNER_DEBUG}" ]]; then set -x; fi
+
+          git config --global --add safe.directory "${GITHUB_WORKSPACE}"
+
+          dnf install -y -q gh
+          gh extension install actions/gh-actions-cache
+
+          cache_key='flatpak-builder-${{ hashFiles('./build-aux/com.obsproject.Studio.json') }}-x86_64'
```

The hash need to cover all manifest JSON files, feel free to use another argument for `hashFiles` it just needs to cover all the files.

```suggestion
          cache_key='flatpak-builder-${{ hashFiles('./build-aux/**/*.json') }}-x86_64'
```